flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/operators/AbstractAsyncStateStreamOperator.java

```diff
@@ -333,6 +335,15 @@ public void processWatermark(Watermark mark) throws Exception {
         asyncExecutionController.processNonRecord(() -> super.processWatermark(mark));
     }
 
+    protected void emitWatermarkDirectly(Watermark mark) throws Exception {
```

Comment:
How about rewriting `processWatermark` in `AbstractAsyncStateStreamOperator.java` instead of exposing the `emitWatermarkDirectly` ?

Reply:
Fixed, recovered `emitWatermarkDirectly ` as origin, and rewrote `processWatermark`. Currently, the async operator won't deal with interruptible watermark.